core/htmlmaker/headings.js

```diff
@@ -8,16 +8,31 @@ fs.readFile(file, function processTemplates (err, contents) {
         });
 
 
-//select sections without first-child h1's
-var noHeadSection = $("section:not(:has(h1:first-child))");
-//select sections' nested elements (<p>s, h1's) with classes or text matching key criteria 
-var ataSection = $("section").find("p.AboutAuthorTextNo-Indentatatx1,p.AboutAuthorTextHeadatah,p.AboutAuthorTextatatx,h1:contains(About the Author)");
-var bobadSection = $("section").find("h1.BOBAdTitlebobt,p.BOBAdTextbobtx")
-var fsSection = $("section").find("p.FrontSalesTitlefst,p.FrontSalesSubtitlefsst,p.FrontSalesQuoteHeadfsqh,p.FrontSalesTextfstx,p.FrontSalesTextNoIndentfstx1,p.FrontSalesQuotefsq,p.FrontSalesQuoteNoIndentfsq1")
-
-//for each section without first-child h1's, create one with .Nonprinting and approporate h1 text
+//select sections without direct-child h1's
+var noHeadSection = $("section:not(:has(>h1))");
+
+//for each section without direct-child h1's, create one with .Nonprinting and approporate h1 text
 noHeadSection.each (function() {
-  var hText = "Frontmatter";
+
+  // the var for new h1 text
+  var hText = ''
+
+  // find the header element
+  var header = $(this).children("header")
+
+  // get text from h1 in the header element
+  var headerText = header.children("h1").text()
+
+  // get text from the section data-type
+  var dataTypeText = $(this).attr("data-type")
```

Comment:
@nelliemckesson I added the sanitization at line 33 for the data-type text (I just borrowed yours).
Tested, it worked!  I'm going to go ahead and merge since I'm not merging into Production anyways, if you see a change you want me to make let me know!  
(Just trying to save an extra back n forth)